src/workflows/multiomics/multisample/main.nf

```diff
@@ -1,155 +1,186 @@
+    multisample_ch = input_ch
+      // The input for this workflow can either be a list of unimodal files
+      // or a single multimodal file. To destingish between the two, the files will be split either way.
+      // For mutliple unimodal files, the result before or after splitting is identical.
+      // In both cases, this workflow requires split files.
+      
+      // Split must be called on each item of the input list, so split it into multiple events with unique ids
+      // Unique ids are required to run a component
+      | flatMap {id, state ->
+        def newEvents = state.input.withIndex().collect{input_file, index -> 
+          def newState = state + ["input": input_file]
+          ["${id}_${index}", newState]
+        }
+        newEvents
       }
-    | merge.run(args: [ output_compression: "gzip" ])
-    | pmap {id, data ->
-        def new_data = ["input": data.output]
-        [id, new_data]
-    }
-    | integration_setup_workflow
-    | pmap {id, data ->
-        def new_data = ["input": data.output]
-        [id, new_data]
-    }
-    | getWorkflowArguments(key: "publish")
-    | publish
-    | map {list -> [list[0], list[1]]}
-
-  emit:
-  output_ch
-}
-
-workflow test_wf {
-  helpMessage(config)
-
-  // allow changing the resources_test dir
-  params.resources_test = params.rootDir + "/resources_test"
-
-  // or when running from s3: params.resources_test = "s3://openpipelines-data/"
-  
-  testParams = [
-      param_list: [
-        [
-          id: "test",
-          input: params.resources_test + "/concat_test_data/concatenated_brain_filtered_feature_bc_matrix_subset.h5mu",
-          publish_dir: "foo/"
-        ],
-        [
-          id: "test2",
-          input: params.resources_test + "/concat_test_data/concatenated_brain_filtered_feature_bc_matrix_subset.h5mu",
-          publish_dir: "foo/"
-        ],
-      ]
-    ]
-
-
-  output_ch =
-    channelFromParams(testParams, config)
-      | view { "Input: $it" }
-      | run_wf
-      | view { output ->
-        assert output.size() == 2 : "outputs should contain two elements; [id, file]"
-        assert output[1].output.toString().endsWith(".h5mu") : "Output file should be a h5mu file. Found: ${output[1]}"
-        "Output: $output"
+      | split_modalities_workflow.run(
+        fromState: {id, state ->
+          [
+            "input": state.input,
+            "id": id
+          ]
+        },
+        toState: [
+          "output": "output", 
+          "output_types": "output_types"
+        ]
+      )
+      // gather the output from split_modalities_workflow
+      // by reading the output csv (the csv contains 1 line per output file)
+      | flatMap {id, state ->
+        def outputDir = state.output
+        def types = readCsv(state.output_types.toUriString())
+        
+        types.collect{ dat ->
+          def reg = ~/_[0-9]+$/
+          def new_id = id - reg + "_${dat.name}" // Make a unique ID again by appending the modality name.
```

Comment:
See above

```suggestion
          def new_id = state.orig_id + "_${dat.name}" // Make a unique ID again by appending the modality name.

Reply:
Fixed in 7ddbf7f3a60fd6d5906eff610c332ae6fefeae78